src/hmem_cuda.c

```diff
@@ -754,12 +725,33 @@ int cuda_host_unregister(void *ptr)
 
 enum cuda_xfer_setting cuda_get_xfer_setting(void)
 {
+	enum cuda_xfer_setting hmem_cuda_xfer_setting;
+	int ret, param_value = 1;
+
+	/* Fetch the value of the environment variable FI_HMEM_CUDA_ENABLE_XFER */
+	ret = fi_param_get_bool(NULL, "hmem_cuda_enable_xfer", &param_value);
+	if (ret == -EINVAL){
+		fprintf(stderr, "FI_HMEM_CUDA_ENABLE_XFER was set to an invalid value by the user."
+			" FI_HMEM_CUDA_ENABLE_XFER is boolean and can be set to only 0/false/no/off or"
+			" 1/true/yes/on.\n");
+		abort();
+	}
+
+	if (ret) {
+		hmem_cuda_xfer_setting = CUDA_XFER_UNSPECIFIED;
+	} else {
+		hmem_cuda_xfer_setting = (param_value > 0) ? CUDA_XFER_ENABLED : CUDA_XFER_DISABLED;
+	}
+
+	FI_INFO(&core_prov, FI_LOG_CORE, "cuda_xfer_setting: %d\n", hmem_cuda_xfer_setting);
+
 	return hmem_cuda_xfer_setting;
 }
 
 bool cuda_is_ipc_enabled(void)
 {
-	return cuda_ipc_enabled;
+	/* CUDA IPC is only enabled if cuda_hmem_init() is successful */
+	return cuda_hmem_devices_verified;
```

Comment:
Is it enough? ipc_enable should not be true when xfer is disabled. Are we changing this behavior?

```
/*
	 * CUDA IPC is only enabled if cudaMemcpy can be used.
	 */
	cuda_ipc_enabled = (hmem_cuda_xfer_setting != CUDA_XFER_DISABLED);
```

Reply:
I think you are looking at the older version of the commit, is it? It is changed in the recent version.

Reply:
But `cuda_hmem_devices_verified` is true as long as cuda_hmem_init succeed, right? It does not depend on xfer settings.

Reply:
@shijin-aws, in one of the comments (The comment is open above in this PR), @wzamazon suggested not to do that check as that would deteriorate the performance of shm provider.

Reply:
I see, I will follow in that thread instead.

Reply:
Yes. this function (`cuda_is_ipc_enabled`) is called every time shm provider send a message, with this change, `cuda_get_xfer_setting()` became expensive because it will call `getenv` and parse the result.

So if we call `cuda_get_xfer_setting()` in `cuda_is_ipc_enabled()`. SHM performance will be impacted.